### PR TITLE
[FIX] rebalance barbaran montante MA

### PR DESCRIPTION
**Pull request by leoCottret**
#### Summary
Balance "Rebalance barbaran montante MA"

#### Purpose of change
Barbaran montante was too powerful at low STR level, because you could reach 300% armor penetration, scaling with strength.
The description was also changed a bit to better reflect historical accuracy. (thanks krwak)
The impaling technique was also a bit useless (10% damage bonus, applied before armor reduction, which usually makes close to no difference).

#### Describe the solution
After a very long discussion, we agreed on a different value: 200% total armor penetration (arpen).
The static bonus is changed from 75% arpen to 100%.
The onpause bonus is changed from 225% arpen to 125%.

Also, the impalade technique is changed from +10% damage to +30% damage.

#### Describe alternatives you've considered
A gazillon, but at some point we need to choose.

#### Testing
Tested with 6 skills in everything, except dodge which I increased or lowered depending on if the MA benefits from it (some styles get buff from blocking so you'll prefer to avoid dodging with those ones). All tests were done against a kevlar hulk. If a MA had a counter attack (on dodge or parry), I added the damage to the next hit (it's essentialy a 0 move cost hit). If I got hurt or too low on stamina, I reseted it through the debug menu. 

The part in bold is the context (eg "BM, 12 STR, sledge" means "Barabaran Montante style, with a character that has 12 strength, with a sledgehammer").
The average damage is just below.

At 12 STR, this will be the best against heavy armored opponents, as it should.
At 18 STR, it's already beaten by Fior, but still pretty decent, be it with a mainly cutting weapon (zweihander) or a bashing weapon (sledgehammer).

This is a best case scenario. Against enemies that don't have armor, this style is one of the worst choice. It has no counter attack on block or dodge, no grab break etc. So this is an anti armor focused style, that is now nerfed at low STR.

image

### CODE REVIEW

File: data/json/martialarts.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "type": "martial_art",
     "id": "style_barbaran",
     "name": { "str": "Barbaran Montante" },
-    "description": "The Barbaran Montante originated in 16th century Spain and focuses on overcoming thick armor with heavy weaponry and sweeping blows, transferring kinetic energy through armor and into vulnerable organs and bones.",
+    "description": "The Barbaran Montante originated in 16th century Spain and focuses on overcoming thick armor with heavy weaponry and sweeping blows, transferring kinetic energy through armor and into vulnerable organs and bones. This style concentrates on the modern use of two handed swords but could be adapted to other types of two handed weaponry.",
```
Comment (chaosvolt) [inline data/json/martialarts.json:71]:
```suggestion
    "description": "The Barbaran Montante originated in 16th century Spain and focuses on overcoming thick armor with heavy weaponry and sweeping blows, transferring kinetic energy through armor and into vulnerable organs and bones.  This style concentrates on the modern use of two handed swords but could be adapted to other types of two handed weaponry.",
```

**Comment by Coolthulhu** [comment]
From your results, it looks like Fior still beats it at high strength, at something that is supposed to be BM's best use case.
This limits the niche that BM can have.

Manual tests like this aren't very informative. It would be much better if you set up automated tests, with brute force dmg averaging.
This way we'd have info not just about one target type, but multiple high level ones. Testing different stat levels and maybe even encumbrance levels would be possible.
It's hard to balance an entire MA based on a couple of manual tests.

It would be good if you had some numbers to compare to. For example:
* BM should deal 20% more dmg than any other MA, using any common melee weapon, when a 12 STR character with 5 in all skills attacks a Kevlar hulk
* BM should not deal more than Fior when a 10 STR character with 10 in all stats attacks a standard Hulk

Without those, the changes look mostly like number fiddling and number fiddling is a sign of lack of design.


I have failed to post this post (closed tab before realizing) like 3 times now, on 3 different days, rewritten from scratch every time. This is the primary reason for late feedback.

**Comment by leoCottret** [comment]
> From your results, it looks like Fior still beats it at high strength, at something that is supposed to be BM's best use case. This limits the niche that BM can have.

I know I said something like that at first but it seems more logical to say that BM fits "low" STR character instead.
As an anti armor style they will always fall back behind a style with bonus damage/extra attacks at some point, because we don't have monsters with ~100 armor (yet?) like DDA do.
So we'll always reach a point where BM will negate more armor than what any monster has, then at this point most styles that boost damage a bit will become better.
Unless we reduce the armor penetration and give it some damage boost instead, in that case it will scale better at high STR.

I think the niche is (at least, right now):
- Low STR characters that want to specialize in an other stats to change a bit and still being able to beat most monsters in melee, with high enough skills. A high STR character is the meta now, this offers an alternative, so it's a step in the right direction, I think.
- People that want a boost in damage at early game, the part where you're the most vulnerable
- And let's not forget, people that want to keep using a non focused bash weapon (like a zweihander), and switch to this style when they meet an opponent with a high armor.

Well, I'll try to set up some automated tests. I can understand the use of automated test to facilitate further MA testing anyway.

**Comment by leoCottret** [comment]
Here's some data I could get with the new MA test system I'm creating.
To summarize:
You have a lot of options to choose (cf top of the function). 
The it selects a weapon compatible with the MA (or not if you choose a specific weapon to test).
Then it spawns a player and a monster, and make them hit each other.
I could make the counter attacks on dodge/block trigger manually (33% of the time). This is important to judge the balance of a MA.
After a few hits, it stores the health that the monster lost, and reset everything.
It does that 80 times (it gives a reasonable precision), then make the "average damage", taking into account the attack cost of the weapon.
I skipped some steps, but that's the basic idea.
So here's some automated tests (you can copy paste the raw result in a markdown table generator).
This is more of a toolbox than a test case to be verified each time, and I'll try to find a way to no be executed each time there's the merge (if it is). That's not the point and it would take too much time.

| Starting tests for -> STATS: 12 SKILLS: 5 M: mon_zombie_kevlar_2 |                       |         |
|------------------------------------------------------------------|-----------------------|---------|
| Fior Di Battaglia                                                | : sledge hammer       | : 21.5  |
| Barbaran Montante                                                | : sledge hammer       | : 20.6  |
| Ninjutsu                                                         | : loaded stick        | : 15.1  |
| Medieval Swordsmanship                                           | : longsword replica   | : 14.7  |
| Niten Ichi-Ryu                                                   | : nodachi replica     | : 14.4  |
| Bionic Combatives                                                | : monomolecular blade | : 2.6   |
| Starting tests for -> STATS: 15 SKILLS: 7 M: mon_zombie_kevlar_2 |                       |         |
| Barbaran Montante                                                | : mace                | : 34.5  |
| Fior Di Battaglia                                                | : sledge hammer       | : 34.1  |
| Niten Ichi-Ryu                                                   | : bokken              | : 28.2  |
| Ninjutsu                                                         | : loaded stick        | : 26.6  |
| Medieval Swordsmanship                                           | : zweihaender replica | : 25.7  |
| Bionic Combatives                                                | : monomolecular blade | : 4.0   |
| Starting tests for -> STATS: 18 SKILLS: 9 M: mon_zombie_kevlar_2 |                       |         |
| Barbaran Montante                                                | : mace                | : 55.1  |
| Fior Di Battaglia                                                | : sledge hammer       | : 51.8  |
| Niten Ichi-Ryu                                                   | : bokken              | : 48.0  |
| Ninjutsu                                                         | : loaded stick        | : 42.6  |
| Medieval Swordsmanship                                           | : zweihaender replica | : 41.5  |
| Bionic Combatives                                                | : bionic claws        | : 15.5  |
| Starting tests for -> STATS: 12 SKILLS: 5 M: mon_zombie_hulk     |                       |         |
| Bionic Combatives                                                | : monomolecular blade | : 80.9  |
| Fior Di Battaglia                                                | : battle axe          | : 62.4  |
| Ninjutsu                                                         | : monomolecular blade | : 61.5  |
| Barbaran Montante                                                | : battle axe          | : 58.0  |
| Niten Ichi-Ryu                                                   | : hand-forged sword   | : 56.6  |
| Medieval Swordsmanship                                           | : broadsword          | : 49.2  |
| Starting tests for -> STATS: 15 SKILLS: 7 M: mon_zombie_hulk     |                       |         |
| Bionic Combatives                                                | : monomolecular blade | : 101.2 |
| Fior Di Battaglia                                                | : battle axe          | : 79.4  |
| Ninjutsu                                                         | : monomolecular blade | : 74.2  |
| Niten Ichi-Ryu                                                   | : hand-forged sword   | : 73.8  |
| Barbaran Montante                                                | : war hammer          | : 72.3  |
| Medieval Swordsmanship                                           | : broadsword          | : 63.7  |
| Starting tests for -> STATS: 18 SKILLS: 9 M: mon_zombie_hulk     |                       |         |
| Bionic Combatives                                                | : monomolecular blade | : 132.0 |
| Barbaran Montante                                                | : war hammer          | : 91.9  |
| Niten Ichi-Ryu                                                   | : hand-forged sword   | : 91.8  |
| Fior Di Battaglia                                                | : battle axe          | : 91.3  |
| Medieval Swordsmanship                                           | : estoc               | : 80.9  |
| Ninjutsu                                                         | : broadsword          | : 78.7  |
|                                                                  |                       |         |

Keeping in mind that this is when you keep hitting an enemy, I couldn't find a way to make him wait and have the onpause debuff (maybe it's simple though), and it may be important to judge this style.
So just looking at those results, I think Barbaran Montante is in a rather good spot right now. I took the best of the MA and it can still compete with them with the static debuff and no counter attacks. Well I won't rant much about it, let me now what you think and how I could improve the game system, if you have some (other) tips @Coolthulhu. 

To finish, my main concerns are:
- Being sure the debuff from onblock, onhit etc. are applied
- Making the character wait until the end of turn, for the onpause debuff
- Having automatic counter attacks instead of triggering them manually, I had some success by changing the moves left for the player
``` cpp
// Important to trigger dodge counters/block counters, having enough moves means block/dodge counters left are reset in process_turn(). But it's far from 100% reliable, it just triggers them rarely instead of never
g->u.mod_moves( 300 );
p.process_turn();
```
- And yes, more generally having a virtual system closer to the game's one without recreating the game.

**Comment by leoCottret** [comment]
I added the test system as a new PR, it's now out of scope of the BM rebalancing and could help focus the discussion on the system and not this MA balance in particular.
Also this became a much bigger project that the several lines modified that this rebalancing does.
https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1921

**Comment by chaosvolt** [comment]
Additional suggestion: add the hand-forged sword (`sword_metal`) to the martial art weapon list. Evidently either Barbaran was added before DDA's version of the "generic handmade sword" idea was merged, or they just forgot.

DDA's version is basically in the same weapon family as the nail sword and friends (despite being vastly more effort to make), while the BN version is balanced more to fit a "heavier but worth the effort as a sidegrade to the survivor machete" role, as per: https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1028

**Comment by leoCottret** [comment]
> Additional suggestion: add the hand-forged sword (`sword_metal`) to the martial art weapon list. Evidently either Barbaran was added before DDA's version of the "generic handmade sword" idea was merged, or they just forgot.
> 
> DDA's version is basically in the same weapon family as the nail sword and friends (despite being vastly more effort to make), while the BN version is balanced more to fit a "heavier but worth the effort as a sidegrade to the survivor machete" role, as per: #1028

Ok since I finally finished the autoit version I could run some tests with some modifications.
This adds a 0.5 on pause cutting buff.
I also added the sword.
Those are the result, on my end I think this PR is ready to be merged.


---------------------------------------------------------------------------------------------------

Both hulks with only longswipe attacks and 9999 hp
iterations 80, fight cycles 3, default armor
Both style with a wait one turn, then attack, for each fight cycle

# NEW TESTS
## STATS: 12, SKILLS: 5, M: Kevlar hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | battle axe                      | 40.87 |
| Barbaran Montante               | mace                            | 36.85 |
| Barbaran Montante               | war hammer                      | 36.47 |
| Barbaran Montante               | zweihaender                     | 32.32 |
| Barbaran Montante               | longsword                       | 32.07 |
| Barbaran Montante               | morningstar                     | 31.86 |
| Barbaran Montante               | battle axe replica              | 29.49 |
| Barbaran Montante               | morningstar fake                | 28.5 |
| Barbaran Montante               | mace replica                    | 28.05 |
| Barbaran Montante               | morningstar replica             | 27.85 |
| Barbaran Montante               | sledge hammer                   | 27.66 |
| Barbaran Montante               | longsword replica               | 25.51 |
| Barbaran Montante               | zweihaender replica             | 22.39 |
| Barbaran Montante               | war flail                       | 22.11 |
| Barbaran Montante               | mace fake                       | 21.61 |
| Fior Di Battaglia               | sledge hammer                   | 20.07 |
| Barbaran Montante               | heavy sledge hammer             | 19.95 |
| Barbaran Montante               | crude sword                     | 19.41 |
| Barbaran Montante               | peasant flail                   | 18.9 |
| Barbaran Montante               | nord                            | 18.54 |
| Barbaran Montante               | 2-by-sword                      | 18.39 |
| Barbaran Montante               | battle axe fake                 | 17.54 |
| Fior Di Battaglia               | battle axe replica              | 16.67 |
| Barbaran Montante               | longsword fake                  | 16.05 |
| Barbaran Montante               | zweihaender fake                | 14.36 |
| Fior Di Battaglia               | lucerne hammer fake             | 12.38 |
| Fior Di Battaglia               | lucerne hammer                  | 11 |
| Fior Di Battaglia               | ji                              | 9.48 |
| Fior Di Battaglia               | short sledge hammer             | 9.44 |
| Fior Di Battaglia               | battle axe                      | 7.84 |
| Fior Di Battaglia               | copper axe                      | 5.31 |
| Fior Di Battaglia               | halberd                         | 4.98 |
| Fior Di Battaglia               | hockey stick                    | 4.83 |
| Fior Di Battaglia               | fire axe                        | 4.53 |
| Fior Di Battaglia               | lobotomizer                     | 4.2 |
| Fior Di Battaglia               | wood axe                        | 3.89 |
| Fior Di Battaglia               | pickaxe                         | 3.67 |
| Fior Di Battaglia               | battle axe fake                 | 3.61 |
| Fior Di Battaglia               | halberd fake                    | 2.74 |
| Fior Di Battaglia               | makeshift glaive                | 2.26 |
| Fior Di Battaglia               | glaive                          | 1.91 |
| Fior Di Battaglia               | hoe                             | 0.49 |
| Fior Di Battaglia               | stone axe                       | 0.15 |
|                                 |                                 |         |
## STATS: 18, SKILLS: 9, M: Kevlar hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | war hammer                      | 93.81 |
| Barbaran Montante               | battle axe                      | 80.64 |
| Barbaran Montante               | morningstar                     | 74.95 |
| Barbaran Montante               | mace                            | 74.92 |
| Barbaran Montante               | battle axe replica              | 69.9 |
| Barbaran Montante               | crude sword                     | 66.56 |
| Barbaran Montante               | longsword                       | 64.42 |
| Barbaran Montante               | zweihaender                     | 62.91 |
| Barbaran Montante               | longsword replica               | 58.23 |
| Barbaran Montante               | sledge hammer                   | 57.85 |
| Barbaran Montante               | morningstar fake                | 56.53 |
| Barbaran Montante               | morningstar replica             | 54.36 |
| Barbaran Montante               | mace replica                    | 52.84 |
| Barbaran Montante               | zweihaender replica             | 50.93 |
| Fior Di Battaglia               | sledge hammer                   | 47.04 |
| Barbaran Montante               | war flail                       | 44.28 |
| Barbaran Montante               | mace fake                       | 43.59 |
| Barbaran Montante               | 2-by-sword                      | 42.11 |
| Fior Di Battaglia               | battle axe replica              | 41.81 |
| Barbaran Montante               | nord                            | 41.74 |
| Barbaran Montante               | heavy sledge hammer             | 38.52 |
| Barbaran Montante               | peasant flail                   | 37.26 |
| Barbaran Montante               | battle axe fake                 | 35.29 |
| Barbaran Montante               | longsword fake                  | 34.45 |
| Fior Di Battaglia               | short sledge hammer             | 31.67 |
| Barbaran Montante               | zweihaender fake                | 30.77 |
| Fior Di Battaglia               | lucerne hammer                  | 30.11 |
| Fior Di Battaglia               | lucerne hammer fake             | 26.45 |
| Fior Di Battaglia               | hockey stick                    | 25.14 |
| Fior Di Battaglia               | battle axe                      | 21.6 |
| Fior Di Battaglia               | copper axe                      | 19.43 |
| Fior Di Battaglia               | fire axe                        | 18.71 |
| Fior Di Battaglia               | ji                              | 17.18 |
| Fior Di Battaglia               | lobotomizer                     | 16.29 |
| Fior Di Battaglia               | halberd                         | 14.7 |
| Fior Di Battaglia               | pickaxe                         | 13.63 |
| Fior Di Battaglia               | wood axe                        | 13.5 |
| Fior Di Battaglia               | battle axe fake                 | 12.18 |
| Fior Di Battaglia               | makeshift glaive                | 11.25 |
| Fior Di Battaglia               | halberd fake                    | 10.42 |
| Fior Di Battaglia               | glaive                          | 10.02 |
| Fior Di Battaglia               | hoe                             | 7.55 |
| Fior Di Battaglia               | stone axe                       | 2.25 |
|                                 |                                 |         |
## STATS: 12, SKILLS: 5, M: zombie hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Fior Di Battaglia               | fire axe                        | 81.83 |
| Fior Di Battaglia               | battle axe                      | 69.33 |
| Fior Di Battaglia               | lobotomizer                     | 65.27 |
| Barbaran Montante               | battle axe                      | 59.64 |
| Barbaran Montante               | war hammer                      | 55.27 |
| Fior Di Battaglia               | makeshift glaive                | 51.96 |
| Fior Di Battaglia               | wood axe                        | 49.96 |
| Barbaran Montante               | longsword                       | 47.84 |
| Fior Di Battaglia               | ji                              | 45.48 |
| Fior Di Battaglia               | halberd                         | 45.26 |
| Barbaran Montante               | zweihaender                     | 44.84 |
| Fior Di Battaglia               | glaive                          | 44.01 |
| Fior Di Battaglia               | battle axe replica              | 43.87 |
| Barbaran Montante               | crude sword                     | 42.88 |
| Fior Di Battaglia               | lucerne hammer                  | 41.96 |
| Fior Di Battaglia               | sledge hammer                   | 41.01 |
| Barbaran Montante               | morningstar                     | 41.01 |
| Barbaran Montante               | battle axe replica              | 39.66 |
| Barbaran Montante               | mace                            | 38.9 |
| Fior Di Battaglia               | hockey stick                    | 35.17 |
| Fior Di Battaglia               | short sledge hammer             | 35 |
| Barbaran Montante               | longsword replica               | 34.35 |
| Barbaran Montante               | morningstar replica             | 33.33 |
| Barbaran Montante               | sledge hammer                   | 31.77 |
| Barbaran Montante               | morningstar fake                | 31.66 |
| Barbaran Montante               | mace replica                    | 31.64 |
| Barbaran Montante               | zweihaender replica             | 31.02 |
| Fior Di Battaglia               | copper axe                      | 29.35 |
| Fior Di Battaglia               | pickaxe                         | 27.91 |
| Barbaran Montante               | nord                            | 24.65 |
| Barbaran Montante               | mace fake                       | 23.75 |
| Barbaran Montante               | war flail                       | 23.06 |
| Barbaran Montante               | 2-by-sword                      | 22.91 |
| Barbaran Montante               | peasant flail                   | 22.69 |
| Barbaran Montante               | heavy sledge hammer             | 22.22 |
| Fior Di Battaglia               | lucerne hammer fake             | 19.27 |
| Fior Di Battaglia               | battle axe fake                 | 18.62 |
| Barbaran Montante               | battle axe fake                 | 18.32 |
| Barbaran Montante               | longsword fake                  | 18.25 |
| Barbaran Montante               | zweihaender fake                | 17.69 |
| Fior Di Battaglia               | hoe                             | 16.34 |
| Fior Di Battaglia               | halberd fake                    | 14.82 |
| Fior Di Battaglia               | stone axe                       | 8.36 |
|                                 |                                 |         |
## STATS: 18, SKILLS: 9, M: zombie hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Fior Di Battaglia               | fire axe                        | 92.45 |
| Barbaran Montante               | war hammer                      | 91.21 |
| Fior Di Battaglia               | battle axe                      | 82.41 |
| Barbaran Montante               | battle axe                      | 82.25 |
| Barbaran Montante               | morningstar                     | 75.24 |
| Fior Di Battaglia               | lobotomizer                     | 74.82 |
| Barbaran Montante               | mace                            | 74.34 |
| Fior Di Battaglia               | makeshift glaive                | 74.29 |
| Barbaran Montante               | battle axe replica              | 70.02 |
| Fior Di Battaglia               | battle axe replica              | 69.31 |
| Fior Di Battaglia               | lucerne hammer                  | 68.21 |
| Barbaran Montante               | crude sword                     | 65.87 |
| Barbaran Montante               | longsword                       | 65.86 |
| Barbaran Montante               | zweihaender                     | 63.51 |
| Fior Di Battaglia               | sledge hammer                   | 62.83 |
| Fior Di Battaglia               | wood axe                        | 60.29 |
| Fior Di Battaglia               | hockey stick                    | 59.98 |
| Fior Di Battaglia               | halberd                         | 58.8 |
| Fior Di Battaglia               | ji                              | 58.48 |
| Barbaran Montante               | sledge hammer                   | 58.36 |
| Barbaran Montante               | morningstar fake                | 58.29 |
| Barbaran Montante               | longsword replica               | 57.79 |
| Barbaran Montante               | morningstar replica             | 55.25 |
| Barbaran Montante               | zweihaender replica             | 54.46 |
| Fior Di Battaglia               | glaive                          | 54.02 |
| Barbaran Montante               | mace replica                    | 53.44 |
| Fior Di Battaglia               | short sledge hammer             | 53.29 |
| Fior Di Battaglia               | lucerne hammer fake             | 48.06 |
| Fior Di Battaglia               | pickaxe                         | 48 |
| Fior Di Battaglia               | copper axe                      | 47.93 |
| Barbaran Montante               | war flail                       | 45.1 |
| Barbaran Montante               | mace fake                       | 43.94 |
| Barbaran Montante               | nord                            | 43.19 |
| Barbaran Montante               | 2-by-sword                      | 42.28 |
| Barbaran Montante               | heavy sledge hammer             | 41.33 |
| Barbaran Montante               | peasant flail                   | 39.14 |
| Barbaran Montante               | battle axe fake                 | 35.5 |
| Barbaran Montante               | longsword fake                  | 34.5 |
| Fior Di Battaglia               | battle axe fake                 | 32.9 |
| Barbaran Montante               | zweihaender fake                | 30.52 |
| Fior Di Battaglia               | hoe                             | 29.03 |
| Fior Di Battaglia               | halberd fake                    | 26.75 |
| Fior Di Battaglia               | stone axe                       | 18.88 |
|                                 |                                 |       |

**Comment by leoCottret** [comment]
As I said on Discord, last changes on my end for this MA. Feel free to merge it, change it or not.
To balance it and keep it arpen based, the main problem is that bashing damage and worse piercing damage, have already a big innate armor piercing abilities in comparison to cutting.
Also we're comparing with 18 STR max for reasons, when an end game character that focus on STR can, at least temporarily, have way more, and that's where damage boost bonuses will beat arpen (that will be "capped" by the monster armor sooner or later).
Keeping that in mind and to avoid "tweak changes", the piercing values are now as follow:

Static bash: 0.75
Static pierce: 0.75
Static cut: 0.75
Onpause bash: 1.25
Onpause pierce: 1.25
Onpause cut: 2.25

The results, selecting a few of the best weapons for each MA:

# NEW TESTS STARTING AT - 06/10/2022 19:46:06
**Fight cycle method: Wait then hit, Iterations: 80, Fight cycles: 3**
## STATS: 12, SKILLS: 5, M: Kevlar hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | battle axe                      | 42.76 |
| Barbaran Montante               | zweihaender                     | 32.77 |
| Barbaran Montante               | war hammer                      | 33.19 |
| Barbaran Montante               | mace                            | 33.11 |
| Barbaran Montante               | morningstar                     | 29.78 |
| Barbaran Montante               | battle axe replica              | 29.67 |
| Fior Di Battaglia               | sledge hammer                   | 21.6 |
| Fior Di Battaglia               | battle axe                      | 8.23 |
| Fior Di Battaglia               | fire axe                        | 5.58 |
| Fior Di Battaglia               | lobotomizer                     | 4.82 |
| Fior Di Battaglia               | makeshift glaive                | 2.19 |
|                                 |                                 |       |
## STATS: 18, SKILLS: 9, M: Kevlar hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | war hammer                      | 83.61 |
| Barbaran Montante               | battle axe                      | 81.19 |
| Barbaran Montante               | mace                            | 73.48 |
| Barbaran Montante               | battle axe replica              | 70.83 |
| Barbaran Montante               | morningstar                     | 70.45 |
| Barbaran Montante               | zweihaender                     | 62.39 |
| Fior Di Battaglia               | sledge hammer                   | 47.57 |
| Fior Di Battaglia               | battle axe                      | 23.39 |
| Fior Di Battaglia               | fire axe                        | 14.97 |
| Fior Di Battaglia               | lobotomizer                     | 13.09 |
| Fior Di Battaglia               | makeshift glaive                | 7.19 |
|                                 |                                 |       |
## STATS: 12, SKILLS: 5, M: zombie hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Fior Di Battaglia               | battle axe                      | 69.05 |
| Fior Di Battaglia               | fire axe                        | 61.91 |
| Fior Di Battaglia               | lobotomizer                     | 59.92 |
| Barbaran Montante               | battle axe                      | 56.35 |
| Barbaran Montante               | war hammer                      | 54.56 |
| Barbaran Montante               | zweihaender                     | 45.47 |
| Fior Di Battaglia               | sledge hammer                   | 43.33 |
| Barbaran Montante               | morningstar                     | 41.29 |
| Barbaran Montante               | battle axe replica              | 39.18 |
| Barbaran Montante               | mace                            | 38.45 |
| Fior Di Battaglia               | makeshift glaive                | 35.82 |
|                                 |                                 |       |
## STATS: 18, SKILLS: 9, M: zombie hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | war hammer                      | 91.82 |
| Fior Di Battaglia               | battle axe                      | 83.68 |
| Barbaran Montante               | battle axe                      | 80.96 |
| Barbaran Montante               | morningstar                     | 78.13 |
| Fior Di Battaglia               | lobotomizer                     | 74.19 |
| Barbaran Montante               | mace                            | 73.99 |
| Fior Di Battaglia               | fire axe                        | 73.45 |
| Barbaran Montante               | battle axe replica              | 70.17 |
| Barbaran Montante               | zweihaender                     | 63.68 |
| Fior Di Battaglia               | sledge hammer                   | 63.48 |
| Fior Di Battaglia               | makeshift glaive                | 49.57 |
|                                 |                                 |       |





**Fight cycle method: Hit only, Iterations: 80, Fight cycles: 3**
## STATS: 12, SKILLS: 5, M: Kevlar hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | mace                            | 23.07 |
| Fior Di Battaglia               | sledge hammer                   | 22.12 |
| Barbaran Montante               | battle axe replica              | 18.38 |
| Barbaran Montante               | morningstar                     | 16.84 |
| Barbaran Montante               | battle axe                      | 9.82 |
| Barbaran Montante               | war hammer                      | 8.93 |
| Fior Di Battaglia               | battle axe                      | 8.81 |
| Barbaran Montante               | zweihaender                     | 7.16 |
| Fior Di Battaglia               | lobotomizer                     | 4.63 |
| Fior Di Battaglia               | fire axe                        | 4.32 |
| Fior Di Battaglia               | makeshift glaive                | 1.7 |
|                                 |                                 |       |
## STATS: 18, SKILLS: 9, M: Kevlar hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | mace                            | 54.57 |
| Barbaran Montante               | morningstar                     | 49.12 |
| Barbaran Montante               | battle axe replica              | 47.37 |
| Fior Di Battaglia               | sledge hammer                   | 41.89 |
| Barbaran Montante               | war hammer                      | 35.95 |
| Barbaran Montante               | battle axe                      | 34.88 |
| Barbaran Montante               | zweihaender                     | 25.88 |
| Fior Di Battaglia               | battle axe                      | 19.8 |
| Fior Di Battaglia               | fire axe                        | 13.75 |
| Fior Di Battaglia               | lobotomizer                     | 11.7 |
| Fior Di Battaglia               | makeshift glaive                | 8.16 |
|                                 |                                 |       |
## STATS: 12, SKILLS: 5, M: zombie hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Fior Di Battaglia               | battle axe                      | 68.38 |
| Fior Di Battaglia               | lobotomizer                     | 60.36 |
| Barbaran Montante               | war hammer                      | 57.04 |
| Barbaran Montante               | battle axe                      | 56.55 |
| Fior Di Battaglia               | fire axe                        | 56.45 |
| Fior Di Battaglia               | sledge hammer                   | 43.05 |
| Barbaran Montante               | zweihaender                     | 42.67 |
| Barbaran Montante               | morningstar                     | 42.27 |
| Barbaran Montante               | mace                            | 38.58 |
| Barbaran Montante               | battle axe replica              | 37.52 |
| Fior Di Battaglia               | makeshift glaive                | 32.37 |
|                                 |                                 |       |
## STATS: 18, SKILLS: 9, M: zombie hulk
| Martial Art Name                | Weapon                          | Avg dmg |
|---------------------------------|---------------------------------|---------|
| Barbaran Montante               | war hammer                      | 88.17 |
| Barbaran Montante               | battle axe                      | 79.85 |
| Barbaran Montante               | morningstar                     | 77.35 |
| Fior Di Battaglia               | battle axe                      | 76.03 |
| Barbaran Montante               | mace                            | 73.35 |
| Fior Di Battaglia               | lobotomizer                     | 72.7 |
| Barbaran Montante               | battle axe replica              | 71.42 |
| Fior Di Battaglia               | fire axe                        | 68.75 |
| Barbaran Montante               | zweihaender                     | 62.96 |
| Fior Di Battaglia               | sledge hammer                   | 61.11 |
| Fior Di Battaglia               | makeshift glaive                | 48.28 |
|                                 |                                 |       |

**Comment by Coolthulhu** [comment]
Alright. I don't like the test method used, but it is documented well enough that the chance of it being wrong is low.
The number changes are significant enough, while not being overwhelming in the results.

One thing I really don't like is that impale AoE attack is serving not just as AoE, but also as damage output. The problem here is that it's the least reliable kinds of AoE. But it's not big enough of a problem to block the PR.

**Pull request merged**